### PR TITLE
1287 Do not show filter panel for modules without datanames

### PR DESCRIPTION
**Pull request by m7pr**
Part of https://github.com/insightsengineering/teal/pull/1253
Fixes https://github.com/insightsengineering/teal/issues/1287

This PR do not show `filter-panel` nor `data-summary` if a module has `datanames` set to NULL.

Tested with 

<details><summary> `tm_front_page(show_metadata = FALSE)` </summary>

```
library(teal.modules.general)
data <- teal_data()
data <- within(data, {
  require(nestcolor)
  ADSL <- rADSL
  attr(ADSL, "metadata") <- list("Author" = "NEST team", "data_source" = "synthetic data")
})
datanames(data) <- "ADSL"
join_keys(data) <- default_cdisc_join_keys[datanames(data)]

table_1 <- data.frame(Info = c("A", "B"), Text = c("A", "B"))
table_2 <- data.frame(`Column 1` = c("C", "D"), `Column 2` = c(5.5, 6.6), `Column 3` = c("A", "B"))
table_3 <- data.frame(Info = c("E", "F"), Text = c("G", "H"))

table_input <- list(
  "Table 1" = table_1,
  "Table 2" = table_2,
  "Table 3" = table_3
)

app <- init(
  data = data,
  modules = modules(
    tm_front_page(
      header_text = c(
        "Important information" = "It can go here.",
        "Other information" = "Can go here."
      ),
      tables = table_input,
      additional_tags = HTML("Additional HTML or shiny tags go here <br>"),
      footnotes = c("X" = "is the first footnote", "Y is the second footnote"),
      show_metadata = FALSE
    )
  ),
  header = tags$h1("Sample Application"),
  footer = tags$p("Application footer"),
)

if (interactive()) {
  shinyApp(app$ui, app$server)
}
````

</details>

<details><summary> `tm_front_page(show_metadata = TRUE)` </summary>

```
library(teal.modules.general)
data <- teal_data()
data <- within(data, {
  require(nestcolor)
  ADSL <- rADSL
  attr(ADSL, "metadata") <- list("Author" = "NEST team", "data_source" = "synthetic data")
})
datanames(data) <- "ADSL"
join_keys(data) <- default_cdisc_join_keys[datanames(data)]

table_1 <- data.frame(Info = c("A", "B"), Text = c("A", "B"))
table_2 <- data.frame(`Column 1` = c("C", "D"), `Column 2` = c(5.5, 6.6), `Column 3` = c("A", "B"))
table_3 <- data.frame(Info = c("E", "F"), Text = c("G", "H"))

table_input <- list(
  "Table 1" = table_1,
  "Table 2" = table_2,
  "Table 3" = table_3
)

app <- init(
  data = data,
  modules = modules(
    tm_front_page(
      header_text = c(
        "Important information" = "It can go here.",
        "Other information" = "Can go here."
      ),
      tables = table_input,
      additional_tags = HTML("Additional HTML or shiny tags go here <br>"),
      footnotes = c("X" = "is the first footnote", "Y is the second footnote"),
      show_metadata = TRUE
    )
  ),
  header = tags$h1("Sample Application"),
  footer = tags$p("Application footer"),
)

if (interactive()) {
  shinyApp(app$ui, app$server)
}
````

</details>

<details><summary> `tm_a_pca` </summary>

```
library(teal.modules.general)
library(teal.widgets)

# general data example
data <- teal_data()
data <- within(data, {
  require(nestcolor)
  USArrests <- USArrests
})

datanames(data) <- "USArrests"

app <- init(
  data = data,
  modules = modules(
    tm_a_pca(
      "PCA",
      dat = data_extract_spec(
        dataname = "USArrests",
        select = select_spec(
          choices = variable_choices(
            data = data[["USArrests"]], c("Murder", "Assault", "UrbanPop", "Rape")
          ),
          selected = c("Murder", "Assault"),
          multiple = TRUE
        ),
        filter = NULL
      ),
      ggplot2_args = ggplot2_args(
        labs = list(subtitle = "Plot generated by PCA Module")
      )
    )
  )
)
if (interactive()) {
  shinyApp(app$ui, app$server)
}

````

</details>

# `tm_front_page(show_metadata = FALSE)`

no_metdata

# `tm_front_page(show_metadata = TRUE)`

For `show_metadata = TRUE` the `datanames` are set to `all` and the datasets are shown, as they are also visible in the `Show metadata` modal

metadata

# `tm_a_pca`

image

### CODE REVIEW

**Comment by m7pr** [comment]
Vedha reported that he sees 2 issues - will try to figure this out

```r
── Skipped tests (2) ───────────────────────────────────────────────────────────────────────────
• need a fix in a .slicesGlobal (1): test-module_teal.R:1075:11
• To decide (1): test-init.R:60:3

── Failed tests ────────────────────────────────────────────────────────────────────────────────
Error (test-shinytest2-modules.R:85:3): e2e: filter panel is not displayed when datanames is NULL
Error in `UseMethod("read_xml")`: no applicable method for 'read_xml' applied to an object of class "NULL"
Backtrace:
     ▆
  1. ├─testthat::expect_true(...) at test-shinytest2-modules.R:85:3
  2. │ └─testthat::quasi_label(enquo(object), label, arg = "object")
  3. │   └─rlang::eval_bare(expr, quo_get_env(quo))
  4. ├─rvest::html_attr(...)
  5. │ └─xml2::xml_attr(x, name, default = default)
  6. ├─rvest::html_element(...)
  7. └─app$get_html_rvest(".teal_secondary_col")
  8.   ├─rvest::read_html(self$get_html(selector)) at teal/R/TealAppDriver.R:608:7
  9.   └─xml2:::read_html.default(self$get_html(selector))
 10.     ├─base::suppressWarnings(...)
 11.     │ └─base::withCallingHandlers(...)
 12.     └─xml2::read_xml(x, encoding = encoding, ..., as_html = TRUE, options = options)

Error (test-shinytest2-reporter.R:109:3): e2e: reporter does not show the secondary column that shows filter_panel
Error in `app_find_node_id(self, private, input = input, output = output, 
    selector = selector)`: Cannot find HTML element with selector #teal-teal_modules-example_teal_module-module-reporter-add_report_card_simple-add_report_card_button.shiny-bound-input
Backtrace:
    ▆
 1. └─app$click(NS(app$active_module_ns(), "reporter-add_report_card_simple-add_report_card_button")) at test-shinytest2-reporter.R:109:3
 2.   └─super$click(...) at teal/R/TealAppDriver.R:112:7
 3.     └─shinytest2:::app_click(...)
 4.       └─shinytest2:::app_find_node_id(self, private, input = input, output = output, selector = selector)
 5.         └─shinytest2:::app_abort(...)
 6.           └─rlang::abort(..., app = self, call = call)

[ FAIL 2 | WARN 2 | SKIP 2 | PASS 539 ]
```

File: tests/testthat/test-shinytest2-reporter.R [hunk 1]
```diff
@@ -53,10 +53,6 @@ testthat::test_that("e2e: adding a report card in a module adds it in the report
 
   app$click(NS(app$active_module_ns(), "reporter-add_report_card_simple-add_report_card_button"))
 
-  app$set_input(
-    NS(app$active_module_ns(), "reporter-add_report_card_simple-label"),
-    "Card name"
-  )
```
Comment (m7pr) [inline tests/testthat/test-shinytest2-reporter.R:59]: This was duplicated in the next line, so I deleted one duplication

File: tests/testthat/test-shinytest2-modules.R [hunk 2]
```diff
@@ -72,27 +72,6 @@ testthat::test_that("e2e: filter panel shows all the datasets when datanames is
   app$stop()
 })
 
-testthat::test_that("e2e: filter panel is not displayed when datanames is NULL", {
-  # TODO - this is not satisfied
```
Comment (m7pr) [inline tests/testthat/test-shinytest2-modules.R:76]: This is now satisfied but covered in other tests in `test-shinytest2-reporter.R`

**Comment by m7pr** [comment]
@vedhav I was able to cleanup last 2 tests. Would you mind taking one more look?

**Review by gogonzo — COMMENTED** [review]
This adresses the issue. Please merge once you resolve tests discussion

**Review by vedhav — APPROVED** [review]
LGTM!
There is one small thing that I would have liked but it is out of scope of this PR.
The disappearance of the `filter_hamburger` when the filter panel is not rendered. Because, we follow a design rule that when an icon is present it is actionable by the user.

However, we have the design refactor and we can implement it during that time. If we would like to achieve this right now the easiest way I can think of is to toggle the hamburge by observing the `input$active_tab` and checking if the filter UI exists on the tab.

**Comment by m7pr** [comment]
Hey @vedhav thanks for the comment. I will start a new NICE TO HAVE issue that we can tackle in some free time. Currently I am merging this one

**Comment by m7pr** [comment]
Issue related to the hamburger https://github.com/insightsengineering/teal/issues/1295

**Pull request merged**